store/src/java/com/zimbra/cs/datasource/imap/MessageChanges.java

```diff
@@ -90,6 +97,28 @@ private MessageChanges findChanges(int changeId) throws ServiceException {
         return this;
     }
 
+    private MessageChanges findChangesForDraft(int changeId, int draftFolderId) throws ServiceException {
```

Comment:
Did we separate this out from findChanges because of the fullSync thing?
* It looks like this method additionally runs after findChanges and the conditional push of partial changes? If not then we need to capture tombstones and add a delete change since Drafts can be deleted.
* It looks like we don't care about comparing changeId in here now, so this always runs?

Reply:
The code is a reference from findChanges() only and removed the tombstones check as it's not required here to check for that

The deletion of the draft if any are taken care of before steps, hence removing the tombstones code and check from this as it would be repeatability of same. 
In this, we are only bothered about the draft that is locally modified.
It runs in all scenarios to check for drafts that have been modified if there are any modified it return with modified item details else returns empty

Reply:
You're still querying twice, in separate locations? Will you receive _more_ results the second time if things are happening from other threads (e.g. other clients deleting things, modifying things, etc).

Reply:
In ideal scenario or when I was doing testing it never entered findChanges() in syncMesage as `fullSync` remains true in all possible scenarios . but for this we can check for `!fullsync`
If there is any modification happened by other thread still it would sync up those messages which has been modified locally

Reply:
Try non force sync - polling interval.

Reply:
if we happen to call findChanges() which also looks for tombstones , in case of drafts this errors out . hence we need to have a separate method which won't look for tombstone and looks for only modified items findChangesForDraft()